src/components/Tool/FixPhoneNumbers/Contact.test.tsx

```diff
@@ -321,16 +324,28 @@ describe('Fix PhoneNumber Contact', () => {
     });
   });
 
-  it('should submit form with errors', async () => {
+  it('should show error on each keystroke', async () => {
     const { getByTestId, getAllByTestId } = render(<TestComponent />);
     const textInput = getByTestId('textfield-contactTestId-number2');
     userEvent.clear(textInput);
-    userEvent.type(textInput, 'pq');
+    userEvent.type(textInput, 'p');
 
     await waitFor(() => {
       expect(getAllByTestId('statusSelectError')[1]).toHaveTextContent(
         'This field is not a valid phone number',
       );
     });
   });
+
+  it('should not submit form with errors', async () => {
+    const { getByTestId, getByRole } = render(<TestComponent />);
+    const textInput = getByTestId('textfield-contactTestId-number2');
+    userEvent.clear(textInput);
+    userEvent.type(textInput, 'p');
+
+    userEvent.click(getByRole('button', { name: 'Confirm' }));
+    await waitFor(() => {
+      expect(handleSingleConfirm).not.toHaveBeenCalled();
+    });
+  });
```

Comment:
It might also be worth writing a test to make sure we don't have stale data. But don't waste too much time on it